editoast/src/views/v2/path/path_item_cache.rs

```diff
@@ -0,0 +1,258 @@
+use crate::core::v2::pathfinding::PathfindingResult;
+use crate::error::Result;
+use crate::modelsv2::TrackSectionModel;
+use crate::RetrieveBatchUnchecked;
+use editoast_schemas::infra::TrackOffset;
+use std::collections::HashMap;
+use std::collections::HashSet;
+
+use editoast_models::DbConnection;
+use editoast_schemas::train_schedule::PathItemLocation;
+
+use crate::modelsv2::OperationalPointModel;
+
+type TrackOffsetResult = std::result::Result<Vec<Vec<TrackOffset>>, TrackOffsetExtractionError>;
+
+/// Gather information about several path items, factorizing db calls.
+pub struct PathItemCache {
+    uic_to_ops: HashMap<i64, Vec<OperationalPointModel>>,
+    trigram_to_ops: HashMap<String, Vec<OperationalPointModel>>,
+    ids_to_ops: HashMap<String, OperationalPointModel>,
+    valid_tracks: HashSet<String>,
+}
+
+impl PathItemCache {
+    /// Load the path item cache from a list of pathfinding inputs
+    pub async fn load(
+        conn: &mut DbConnection,
+        infra_id: i64,
+        path_items: &[&PathItemLocation],
+    ) -> Result<PathItemCache> {
+        let (trigrams, ops_uic, ops_id) = collect_path_item_ids(path_items);
+        let uic_to_ops = retrieve_op_from_uic(conn, infra_id, &ops_uic).await?;
+        let trigram_to_ops = retrieve_op_from_trigrams(conn, infra_id, &trigrams).await?;
+        let ids_to_ops = retrieve_op_from_ids(conn, infra_id, &ops_id).await?;
+
+        let tracks = path_items.iter().filter_map(|item| match item {
+            PathItemLocation::TrackOffset(TrackOffset { track, .. }) => {
+                Some((infra_id, track.0.clone()))
+            }
+            _ => None,
+        });
+        let tracks = ids_to_ops
+            .values()
+            .chain(trigram_to_ops.values().flatten())
+            .chain(uic_to_ops.values().flatten())
+            .flat_map(|op| &op.parts)
+            .map(|part| (infra_id, part.track.0.clone()))
+            .chain(tracks);
+        let valid_tracks = TrackSectionModel::retrieve_batch_unchecked::<_, Vec<_>>(conn, tracks)
+            .await?
+            .into_iter()
+            .map(|track| track.obj_id)
+            .collect();
+
+        Ok(PathItemCache {
+            uic_to_ops,
+            trigram_to_ops,
+            ids_to_ops,
+            valid_tracks,
+        })
+    }
+
+    /// Get the operational points associated with an identifier
+    pub fn get_id(&self, id: &String) -> Option<&OperationalPointModel> {
```

Comment:
It's weird to call it `get_id` because you don't get an identifier as a return value. So I suggest changing it like this.
```suggestion
    pub fn get_from_id(&self, id: &String) -> Option<&OperationalPointModel> {
```